src/transactions/ChangeTrustOpFrame.cpp

```diff
@@ -17,6 +17,115 @@
 namespace stellar
 {
 
+static void
+decrementPoolUseCount(AbstractLedgerTxn& ltx, Asset const& asset,
```

Comment:
Should I use nested `LedgerTxn` for safety? I don’t think it’s necessary but wanted to highlight this.

Reply:
I tried to do this but `ChangeTrustOp` stores a `LedgerTxnHeader` at the top and it gets deactivated by the nested `LedgerTxn`. I should be able to work around this though. I'll work through this tomorrow.

Reply:
My last commit adds the nested LedgerTxns. The header wasn't really an issue, but the `trustLine.erase()` was a little messy to deal with. Let me know what you think.